middleware/dnstap/msg/msg.go

```diff
@@ -87,3 +95,29 @@ func (d *Data) ToClientQuery() *tap.Message {
 		QueryPort:      &d.Port,
 	}
 }
+
+// Transform the data into a forwader or resolver query message.
```

Comment:
If it really usefull we should probably put it somewhere under middleware/pkg/...